src/compiler/gcc.rs

```diff
@@ -293,13 +294,18 @@ where
                 };
             }
             Some(Output(p)) => output_arg = Some(p.clone()),
-            Some(NeedDepTarget) => need_explicit_dep_target = true,
+            Some(NeedDepTarget) => {
+                need_explicit_dep_target = true;
+                if need_explicit_dep_argument_path.is_none() {
+                    need_explicit_dep_argument_path = Some(true);
+                }
+            }
             Some(DepTarget(s)) => {
                 dep_flag = OsString::from(arg.flag_str().expect("Dep target flag expected"));
                 dep_target = Some(s.clone());
             }
-            Some(DepArgumentPath(_))
-            | Some(ExtraHashFile(_))
+            Some(DepArgumentPath(_)) => need_explicit_dep_argument_path = Some(false),
```

Comment:
Nit: `need_explicit_dep_argument_path` might be better represented as an enum:
```
enum DepArgumentPathRequirement {
    NotNeeded,
    Needed,
    AlreadyProvided,
}
```
(Feel free to take an artistic liberty to the enum value names, I'm not sure that my suggestion here is the ideal naming 😉)